ui/apps/platform/src/Containers/BaseImages/BaseImagesModal.test.tsx

```diff
@@ -0,0 +1,175 @@
+                        baseImageRepoPath: 'ubuntu',
+                        baseImageTagPattern: '22.04',
+                    },
+                    expect.any(Object)
+                );
+            });
+        });
+
+        it('should parse image path with multiple colons (registry with port)', async () => {
+            render(<BaseImagesModal {...defaultProps} />);
+
+            const input = screen.getByPlaceholderText(/example-registry.io/);
+            const saveButton = screen.getByRole('button', { name: 'Save' });
+
+            fireEvent.change(input, {
+                target: { value: 'docker.io:5000/library/ubuntu:22.04' },
+            });
+            fireEvent.blur(input);
+            fireEvent.click(saveButton);
+
+            await waitFor(() => {
+                expect(mockMutate).toHaveBeenCalledWith(
+                    {
+                        baseImageRepoPath: 'docker.io:5000/library/ubuntu',
+                        baseImageTagPattern: '22.04',
+                    },
+                    expect.any(Object)
+                );
+            });
+        });
+
+        it('should parse image path with tag pattern', async () => {
+            render(<BaseImagesModal {...defaultProps} />);
+
+            const input = screen.getByPlaceholderText(/example-registry.io/);
+            const saveButton = screen.getByRole('button', { name: 'Save' });
+
+            fireEvent.change(input, { target: { value: 'docker.io/library/ubuntu:1.*' } });
+            fireEvent.blur(input);
+            fireEvent.click(saveButton);
+
+            await waitFor(() => {
+                expect(mockMutate).toHaveBeenCalledWith(
+                    {
+                        baseImageRepoPath: 'docker.io/library/ubuntu',
+                        baseImageTagPattern: '1.*',
+                    },
+                    expect.any(Object)
+                );
+            });
+        });
+    });
+
+    describe('form validation', () => {
+        it('should show error when field is empty after blur', async () => {
+            render(<BaseImagesModal {...defaultProps} />);
+
+            const input = screen.getByPlaceholderText(/example-registry.io/);
+
+            fireEvent.focus(input);
+            fireEvent.blur(input);
+
+            await waitFor(() => {
+                expect(screen.getByText('Base image path is required')).toBeInTheDocument();
+            });
+        });
+
+        it('should enable save button when form is valid', async () => {
+            render(<BaseImagesModal {...defaultProps} />);
+
+            const input = screen.getByPlaceholderText(/example-registry.io/);
+            const saveButton = screen.getByRole('button', { name: 'Save' });
+
+            fireEvent.change(input, { target: { value: 'ubuntu:22.04' } });
+
+            await waitFor(() => {
+                expect(saveButton).toBeEnabled();
+            });
+        });
+    });
+
+    describe('submission flow', () => {
```

Comment:
These are good too - but we should check back here after cypress integration tests are added. I suspect that these code paths will be tested more than once. (Whether or not that is a good or bad thing, I'm not sure.)

Reply:
What do you think about testing the different paths and checking if the rendered alert is correct instead of these tests?

```
Mocked success -> Success alert with particular text

Mocked error -> Error alert with particular text
```

I wanted to run a test to make sure the rendered alerts appear correctly. Using mocked data makes sense if we're going to control the result for the sake of the test. We could test this with a Cypress component test. An E2E test would have more overhead.